Remove `send_to_socket::Actor`

thomaseizinger opened the pull request:
This removes a level of indirection for actually sending messages which
simplifies the overall handling of the connection and unifies how
heartbeats are sent to the taker. This is important because we don't
want heartbeats to be handled in a special way as we use them to detect
issues with the connection.

May help with https://github.com/itchysats/itchysats/issues/759.

thomaseizinger committed:
Remove `send_to_socket::Actor`

This removes a level of indirection for actually sending messages which
simplifies the overall handling of the connection and unifies how
heartbeats are sent to the taker. This is important because we don't
want heartbeats to be handled in a special way as we use them to detect
issues with the connection.

May help with https://github.com/itchysats/itchysats/issues/759.

bonomat commented on daemon/src/maker_inc_connections.rs:
```diff
@@ -107,7 +105,26 @@ pub struct Actor {
+
+impl Connection {
+    async fn send(&mut self, msg: wire::MakerToTaker) -> Result<()> {
+        let msg_str = msg.to_string();
```
Shall we add a trace here to log what message is being send?

thomaseizinger replied:
Addressed in https://github.com/itchysats/itchysats/pull/927/commits/21b20f6fdbff870393cd2f92a9d2caee9bfda50c.

thomaseizinger committed:
Add `wire` target logs for tracing messages

Activating logs for the `wire` target will trace all incoming and
outgoing messages for both maker and taker.

bonomat commented on daemon/src/maker_inc_connections.rs:
```diff
@@ -118,6 +118,8 @@ impl Connection {
     async fn send(&mut self, msg: wire::MakerToTaker) -> Result<()> {
         let msg_str = msg.to_string();
 
+        tracing::trace!(target = "wire", taker_id = %self.taker, "Sending {}", msg_str);
```
nice, that's useful 👍

bonomat reviewed (approved):
Awesome, thanks.
